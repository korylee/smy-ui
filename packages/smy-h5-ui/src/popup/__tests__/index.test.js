import { mount } from '@vue/test-utils'
import Vue from 'vue'
import SmyPopup from '..'

test('test popup plugin', () => {
  Vue.use(SmyPopup)
  expect(Vue.component(SmyPopup.name)).toBeTruthy()
})

const Wrapper = {
  components: {
    SmyPopup,
  },
  data: () => ({ show: false }),
  template: `<smy-popup :show.sync="show" v-bind="$attrs">default slot content</smy-popup>`,
}

test('test popup show', async () => {
  const wrapper = mount(Wrapper)
  expect(wrapper.find('.smy-popup').isVisible()).toBeFalsy()
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.setData({ show: true })
  expect(wrapper.find('.smy-popup').isVisible()).toBeTruthy()
  expect(wrapper.html()).toMatchSnapshot()
  await wrapper.setData({ show: false })
  expect(wrapper.find('.smy-popup').isVisible()).toBeFalsy()
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test popup z-index', async () => {
  const wrapper = mount(Wrapper)
  await wrapper.setData({ show: true })
  const prevPopupZIndex = window.getComputedStyle(wrapper.find('.smy-popup').element).zIndex
  const prevOverlayZIndex = window.getComputedStyle(wrapper.find('.smy-popup__overlay').element).zIndex

  await wrapper.setData({ show: false })
  await wrapper.setData({ show: true })

  expect(window.getComputedStyle(wrapper.find('.smy-popup').element).zIndex).toBe(String(+prevPopupZIndex + 3))
  expect(window.getComputedStyle(wrapper.find('.smy-popup__overlay').element).zIndex).toBe(
    String(+prevOverlayZIndex + 3)
  )

  wrapper.destroy()
})

test('test popup onOpen & onClose', async () => {
  const onOpen = jest.fn()
  const onClose = jest.fn()
  const Wrapper = {
    components: {
      SmyPopup,
    },
    data: () => ({ show: false }),
    methods: {
      onOpen,
      onClose,
    },
    template: `
      <smy-popup :show.sync="show" @open="onOpen" @close="onClose">defaul slot content</smy-popup>
    `,
  }
  const wrapper = mount(Wrapper)
  await wrapper.setData({ show: true })
  expect(onOpen).toHaveBeenCalledTimes(1)

  await wrapper.setData({ show: false })
  expect(onClose).toHaveBeenCalledTimes(1)

  wrapper.destroy()
})

test('tet popup close on clickOverlay', async () => {
  const onClose = jest.fn()
  const onClickOverlay = jest.fn()

  const Wrapper = {
    components: {
      SmyPopup,
    },
    data: () => ({ show: false, closeOnClickOverlay: true }),
    methods: {
      onClickOverlay,
      onClose,
    },
    template: `
      <smy-popup :show.sync="show" :closeOnClickOverlay="closeOnClickOverlay" @click-overlay="onClickOverlay" @close="onClose">
        defaul slot content
      </smy-popup>
    `,
  }
  const wrapper = mount(Wrapper)
  await wrapper.setData({ show: true })

  await wrapper.find('.smy-popup__overlay').trigger('click')
  expect(onClickOverlay).toHaveBeenCalledTimes(1)
  expect(onClose).toHaveBeenCalledTimes(1)

  await wrapper.setData({ closeOnClickOverlay: false, show: true })
  await wrapper.find('.smy-popup__overlay').trigger('click')
  expect(onClickOverlay).toHaveBeenCalledTimes(2)
  expect(onClose).toHaveBeenCalledTimes(1)

  wrapper.destroy()
})
