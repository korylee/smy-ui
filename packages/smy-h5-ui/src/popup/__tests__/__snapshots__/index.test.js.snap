// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test popup show 1`] = `
"<transition-stub name=\\"smy-fade\\">
  <div role=\\"dialog\\" tabindex=\\"0\\" class=\\"smy--box smy-popup\\" style=\\"z-index: 998; display: none;\\">
    <div class=\\"smy-popup__overlay\\" style=\\"z-index: 999;\\"></div>
    <transition-stub name=\\"smy-pop-center\\">
      <!---->
    </transition-stub>
  </div>
</transition-stub>"
`;

exports[`test popup show 2`] = `
"<transition-stub name=\\"smy-fade\\">
  <div role=\\"dialog\\" tabindex=\\"0\\" class=\\"smy--box smy-popup\\" style=\\"z-index: 1001;\\">
    <div class=\\"smy-popup__overlay\\" style=\\"z-index: 1002;\\"></div>
    <transition-stub name=\\"smy-pop-center\\">
      <div class=\\"smy-popup__content smy-popup__content--center\\" style=\\"z-index: 1003;\\">default slot content</div>
    </transition-stub>
  </div>
</transition-stub>"
`;

exports[`test popup show 3`] = `
"<transition-stub name=\\"smy-fade\\">
  <div role=\\"dialog\\" tabindex=\\"0\\" class=\\"smy--box smy-popup\\" style=\\"z-index: 1001; display: none;\\">
    <div class=\\"smy-popup__overlay\\" style=\\"z-index: 1002;\\"></div>
    <transition-stub name=\\"smy-pop-center\\">
      <!---->
    </transition-stub>
  </div>
</transition-stub>"
`;
